src/rgw/CMakeLists.txt

```diff
@@ -57,112 +30,78 @@ set(librgw_common_srcs
   rgw_auth_s3.cc
   rgw_arn.cc
   rgw_basic_types.cc
-  rgw_bucket.cc
+  rgw_bucket_encryption.cc
+  rgw_bucket_initb.cc
   rgw_bucket_layout.cc
-  rgw_bucket_sync.cc
   rgw_cache.cc
-  rgw_d3n_datacache.cc
   rgw_common.cc
   rgw_compression.cc
+  rgw_coroutine.cc
+  rgw_cr_rest.cc
+  rgw_cr_tools.cc
+  rgw_crypt.cc
+  rgw_crypt_sanitize.cc
   rgw_etag_verifier.cc
   rgw_cors.cc
   rgw_cors_s3.cc
+  rgw_datalog.cc
+  rgw_dencoder.cc
   rgw_env.cc
   rgw_es_query.cc
   rgw_formats.cc
-  rgw_gc.cc
-  rgw_gc_log.cc
   rgw_http_client.cc
+  rgw_iam_policy.cc
+  rgw_json_enc.cc
   rgw_keystone.cc
+  rgw_kmip_client.cc
+  rgw_kms.cc
   rgw_ldap.cc
   rgw_lc.cc
   rgw_lc_s3.cc
-  rgw_lc_tier.cc
+  rgw_lc_tier.cc 
+  rgw_log_backing.cc
+  rgw_lua.cc
+  rgw_lua_utils.cc
   rgw_metadata.cc
   rgw_multi.cc
   rgw_multi_del.cc
-  rgw_obj_manifest.cc
-  rgw_pubsub.cc
-  rgw_sync.cc
-  rgw_data_sync.cc
-  rgw_sync_counters.cc
-  rgw_sync_error_repo.cc
-  rgw_sync_module.cc
-  rgw_sync_module_aws.cc
-  rgw_sync_module_es.cc
-  rgw_sync_module_es_rest.cc
-  rgw_sync_module_log.cc
-  rgw_sync_module_pubsub.cc
-  rgw_sync_policy.cc
-  rgw_pubsub_push.cc
-  rgw_notify.cc
   rgw_notify_event_type.cc
-  rgw_sync_module_pubsub_rest.cc
-  rgw_sync_trace.cc
-  rgw_trim_bilog.cc
-  rgw_trim_datalog.cc
-  rgw_trim_mdlog.cc
-  rgw_period_history.cc
-  rgw_period_puller.cc
-  rgw_reshard.cc
-  rgw_coroutine.cc
-  rgw_cr_rados.cc
-  rgw_cr_rest.cc
-  rgw_cr_tools.cc
-  rgw_object_expirer_core.cc
+  rgw_object_lock.cc
+  rgw_oidc_provider.cc
   rgw_op.cc
   rgw_otp.cc
+  rgw_perf_counters.cc
   rgw_policy_s3.cc
   rgw_public_access.cc
   rgw_putobj.cc
-  rgw_putobj_processor.cc
   rgw_quota.cc
-  rgw_rados.cc
   rgw_resolve.cc
   rgw_rest.cc
   rgw_rest_client.cc
   rgw_rest_conn.cc
-  rgw_rest_log.cc
-  rgw_rest_metadata.cc
-  rgw_rest_pubsub.cc
-  rgw_rest_pubsub_common.cc
-  rgw_rest_realm.cc
+  rgw_rest_oidc_provider.cc
   rgw_rest_role.cc
-  rgw_rest_s3.cc
-  rgw_s3select.cc
+  rgw_rest_sts.cc
+  rgw_rest_user_policy.cc
   rgw_role.cc
+  rgw_s3select.cc
   rgw_sal.cc
-  rgw_sal_rados.cc
   rgw_string.cc
+  rgw_sts.cc
+  rgw_sync_error_repo.cc
+  rgw_sync_policy.cc
```

Comment:
sync stuff is below the line

Reply:
there are calls to rgw_sync_pipe_filter::{encode,decode} in rgw_common.cc, rgw_zone.cc, and dbstore/sqlite/sqliteDB.cc
there are calls to rgw_sync_policy_info::{decode_json,dump} in rgw_common.cc and rgw_zone.cc.
there is a calle to rgw_sync_policy_info::dump in rgw_admin.cc

Are these a layering violation, or ???

Reply:
There are calls in rgw_common.cc and rgw_zone.cc to rgw_sync_pipe_filter::{encode,decode}()
There is a call in store/dbstore/sqliteDB.cc to rgw_sync_pipe_filter::decode()
There are calls in rgw_common.cc, rgw_zone.cc, and rgw_sync_policy_info::{decode_json,dump}()
There is a call in grw_admin.cc to rgw_sync_policy_info::dump()

These turn into undefined refs after moving rgw_sync_policy.cc below the line.